packages/compass-aggregations/src/modules/pipeline-builder/pipeline-parser/pipeline-parser.ts

```diff
@@ -99,19 +99,47 @@ type ValidateResponse = {
   errors: PipelineParserError[];
 };
 
+function getOrphanComments(comments: t.Comment[], visitedComments: Set<t.Comment>): t.Comment[] {
+  return comments.filter((comment) => !visitedComments.has(comment));
+}
+
+function getDisabledStagesFromEmptyPipeline(root: t.ArrayExpression) {
+  // Inner comments will only be available here if there is no other
+  // elements in the array, in our case it might mean that all stages
+  // are disabled
+  const isPipelineWithAllDisabledStages = root.elements.length === 0 && root.innerComments?.length;
+  if (!isPipelineWithAllDisabledStages) {
+    return [];
+  }
+
+  const innerComments = root.innerComments as t.Comment[];
+  const [visitedComments, stages] = extractStagesFromComments(innerComments);
+  // As all the stages are disabled and if we have comments that're
+  // at the end of the list, we need to make sure to attach them
+  // to the last stage as trailing comments.
+  const hasOrphanComments = stages.length > 0 && !visitedComments.has(innerComments[innerComments.length - 1]);
+  if (!hasOrphanComments) {
+    root.innerComments = innerComments.filter((node) => {
+      return !visitedComments.has(node);
+    });
+    return stages;
+  }
+
+  const lastStage = stages[stages.length - 1];
+  const orphanComments = getOrphanComments(innerComments, visitedComments);
+  lastStage.trailingComments = orphanComments;
```

Comment:
A few things here catch my eye: this code seems to assume that there will always be a stage returned from parsing the comments, I also think checking for last stage being in the visited list as an indicator for having leftover stages quite confusing: as we know that leading comments will be handled by `extractStagesFromComments`, you have two cases here, either you have at least one stage from comments, and then there might or might not be leftover comments, or you have none and so root comments should stay intact, I can't see a case where innerComments of the root node will be a subset. Based on that I'd maybe suggest to do something like this here:

 ```suggestion
  // No stages extracted from comments, keep the root intact
  if (stages.length === 0)  {
    return stages;
  }
  
  const lastStage = stages[stages.length - 1];
  // Attach all leftover comments to root node
  lastStage.trailingComments = root.innerComments.filter(node => {
    return !visitedComments.has(node);
  });
  // Delete all inner comments from the root node (they are all part of stages now)
  delete root.innerComments;
```

Reply:
i made the changes.